components/org.wso2.carbon.identity.api.server.secret.management/org.wso2.carbon.identity.api.server.secret.management.v1/src/main/java/org/wso2/carbon/identity/api/server/secret/management/v1/core/SecretTypeManagementService.java

```diff
@@ -0,0 +1,256 @@
+        SecretType requestDTO, responseDTO;
+        try {
+            requestDTO = buildSecretTypeRequestDTOFromSecretTypeAddRequest(secretTypeAddRequest);
+            responseDTO = SecretManagementServiceHolder.getSecretConfigManager().addSecretType(requestDTO);
+        } catch (SecretManagementException e) {
+            throw handleSecretMgtException(e, SecretManagementConstants.ErrorMessage.ERROR_CODE_ERROR_ADDING_SECRET,
+                    secretTypeAddRequest.getName());
+        }
+        return buildSecretTypeResponseFromResponseDTO(responseDTO);
+    }
+
+    /**
+     * To create Secret Response object for the post request
+     *
+     * @param secretTypeReq secret object.
+     * @return {@link SecretTypeResponse} .
+     */
+    private SecretTypeResponse buildSecretTypeResponseFromResponseDTO(SecretType secretTypeReq) {
+
+        SecretTypeResponse secretTypeResponse = new SecretTypeResponse();
+        secretTypeResponse.setId(secretTypeReq.getId());
+        secretTypeResponse.setName(secretTypeReq.getName());
+        secretTypeResponse.setDescription(secretTypeReq.getDescription());
+        return secretTypeResponse;
+    }
+
+    /**
+     * Validate the secret post request.
+     *
+     * @param secretTypeAddRequest secret post request.
+     */
+    private void validateSecretTypeAddRequest(SecretTypeAddRequest secretTypeAddRequest) {
+
+        String secretAddName = secretTypeAddRequest.getName();
+        if (StringUtils.isBlank(secretAddName)) {
+            throw handleException(Response.Status.BAD_REQUEST, SecretManagementConstants.ErrorMessage.
+                    ERROR_CODE_REFERENCE_NAME_NOT_SPECIFIED, null);
+        }
+    }
+
+    /**
+     * Build secret requestDTO by secret body request.
+     *
+     * @param secretTypeAddRequest secret type post body.
+     * @return Secret requestDTO object.
+     */
+    private SecretType buildSecretTypeRequestDTOFromSecretTypeAddRequest(SecretTypeAddRequest secretTypeAddRequest) {
+
+        SecretType requestDTO = new SecretType();
+        requestDTO.setName(secretTypeAddRequest.getName());
+        requestDTO.setDescription(secretTypeAddRequest.getDescription());
+        return requestDTO;
+    }
+
+    /**
+     * Delete a secret type by name.
+     *
+     * @param secretTypeName Name of the secret type.
+     */
+    public void deleteSecretType(String secretTypeName) {
+
+        try {
+            SecretManagementServiceHolder.getSecretConfigManager().deleteSecretType(secretTypeName);
+        } catch (SecretManagementException e) {
+            throw handleSecretMgtException(e, SecretManagementConstants.ErrorMessage.
+                    ERROR_CODE_ERROR_DELETING_SECRET, secretTypeName);
+        }
+    }
+
+    /**
+     * Retrieve the secret details by name.
+     *
+     * @param secretTypeName secret name.
+     * @return secret.
+     */
+    public SecretTypeResponse getSecretType(String secretTypeName) {
+
+        try {
+
```

Comment:
Let's remove this new line.

Reply:
Addressed with commit [a6f4632](https://github.com/wso2/identity-api-server/pull/315/commits/a6f463218febb07c19f260f0aa26e33fdec47de3)